frontend/viewer/src/lib/parts-of-speech.ts

```diff
@@ -1,13 +1,13 @@
 ﻿import {derived, type Readable, type Writable, writable} from 'svelte/store';
-import type {IPartOfSpeech, IWritingSystems} from '$lib/dotnet-types';
+import type {IPartOfSpeech} from '$lib/dotnet-types';
 import {useLexboxApi} from './services/service-provider';
-import {pickBestAlternative} from './utils';
+import type {WritingSystemService} from './writing-system-service';
 
 type LabeledPartOfSpeech = IPartOfSpeech & {label: string};
 
 let partsOfSpeechStore: Writable<IPartOfSpeech[] | null> | null = null;
 
-export function usePartsOfSpeech(writingSystemsStore: Readable<IWritingSystems>): Readable<LabeledPartOfSpeech[]> {
+export function usePartsOfSpeech(writingSystemService: WritingSystemService): Readable<LabeledPartOfSpeech[]> {
```

Comment:
rather than passing it in as a parameter can't we just call `useWritingSystemService` in this method?

Reply:
I suppose. Yeah. I'll go with that. I feel like we're supposed to be careful where we call getContext. But it probably doesn't matter as much now that we're not using SvelteKit and `use` in `usePartsOfSpeech` probably suggests that we're in a place that context can be fetched,